src/webgpu/shader/validation/decl/ptr_spelling.spec.ts

```diff
@@ -0,0 +1,157 @@
+export const description = `
+Validate spelling of pointer types.
+
+Pointer types may appear.
+
+They are parameterized by:
+- address space, always
+- store type
+- and access mode, as specified by the table in Address Spaces.
+   Concretely, only 'storage' address space allows it, and allows 'read', and 'read_write'.
+
+A pointer type can be spelled only if it corresponds to a variable that could be
+declared in the program.  So we need to test combinations against possible variable
+declarations.
+`;
+
+// This file tests spelling of the pointer type on let-declared pointers.
+//
+// Spelling of pointer-typed parameters on user-declared functions is tested by
+// webgpu:shader,validation,functions,restrictions:function_parameter_types:"*"
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { AddressSpace, kAccessModeInfo, kAddressSpaceInfo } from '../../types.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+import {
+  pointerType,
+  infoExpander,
+  explicitSpaceExpander,
+  accessModeExpander,
+  getVarDeclShader,
+  supportsWrite,
+  ShaderStage,
+} from './util.js';
+
+// Address spaces that can hold an i32 variable.
+const kNonHandleAddressSpaces = keysOf(kAddressSpaceInfo).filter(
+  as => as !== 'handle'
+) as AddressSpace[];
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('let_ptr_explicit_type_matches_var')
+  .desc(
+    'Let-declared pointer with explicit type initialized from var with same address space and access mode'
+  )
+  .specURL('https://w3.org/TR#ref-ptr-types')
+  .params(u =>
+    u // Generate non-handle variables in all valid permutations of address space and access mode.
+      .combine('addressSpace', kNonHandleAddressSpaces)
+      .expand('info', infoExpander)
+      .expand('explicitSpace', explicitSpaceExpander)
+      .combine('explicitMode', [false, true])
```

Comment:
nit: I had to look up what `explicitMode` meant. I think `explicitAccess` might be clearer.

Reply:
Fixed.